src/server/api/admins.test.ts

```diff
@@ -0,0 +1,158 @@
+import test from 'ava'
+import fastify from 'fastify'
+import sinon from 'sinon'
+import { adminRoutes } from './admins'
+import Store from '../store/index.js'
+import ActivityPubSystem from '../apsystem.js'
+import { ModerationChecker } from '../moderation.js'
+import HookSystem from '../hooksystem'
+import { APIConfig } from '.'
+import { makeSigner } from '../../keypair.js'
+import { generateKeypair } from 'http-signed-fetch'
+
+const mockConfig: APIConfig = {
+  port: 3000,
+  host: 'localhost',
+  storage: 'path/to/storage',
+  publicURL: 'http://localhost:3000'
+}
+
+let server: any
+let mockStore: any
+let mockApsystem: any
+
+test.beforeEach(async () => {
+  server = fastify()
+  mockStore = sinon.createStubInstance(Store)
+
+  mockStore.admins = {
+    list: sinon.stub(),
+    add: sinon.stub(),
+    remove: sinon.stub()
+  }
+  mockStore.admins.list.resolves(['admin1@example.com', 'admin2@example.com'])
+  mockStore.admins.add.resolves()
+  mockStore.admins.remove.resolves()
+
+  mockApsystem = new ActivityPubSystem(mockConfig.publicURL, mockStore, new ModerationChecker(mockStore), new HookSystem(mockStore))
+  sinon.stub(mockApsystem, 'hasAdminPermissionForRequest').resolves(true)
+
+  await adminRoutes(mockConfig, mockStore, mockApsystem)(server)
+})
+
+const simulateSignedRequest = (method: string, path: string): { Signature: string, Date: string } => {
+  const keypair = generateKeypair()
+  const publicKeyId = 'https://example.com/#main-key'
```

Comment:
The key should point at a mocked actor which is in the admins list. e.g. `https://example.com/admin.jsonld#main-key`. We should then have an `actor` at that URL which contains the correct publicKey and preferredUsername fields.

We need to then mock the `webfinger` endpoint to resolve to webfinger data which resolves to the admin's actor URL. https://github.com/RangerMauve/staticpub.mauve.moe/blob/default/.well-known/webfinger